plugins/org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/PropertyUtil.xtend

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xbase.util
+
+import org.eclipse.xtext.common.types.JvmFeature
+import org.eclipse.xtext.common.types.JvmOperation
+
+import static extension java.beans.Introspector.*
+import static extension java.lang.Character.*
+
+/**
+ * @author kosyakov - Initial contribution and API
+ */
+class PropertyUtil {
```

Comment:
Please use the same logic as in org.eclipse.xtext.xbase.scoping.batch.AbstractSessionBasedScope.toProperty(String, JvmFeature, int, int)